integration/integration_test.go

```diff
@@ -1142,6 +1143,150 @@ func testSessionRecordingModes(t *testing.T, suite *integrationTestSuite) {
 	}
 }
 
+// testSyncRecordingFailures given a cluster with recording mode set to
+// sync, ensures the session is terminated in case of session recording
+// failures.
+//
+// To simulate failures, we will start an auth server, proxy and ssh server
+// separated. With the session running, we'll stop the auth server, causing any
+// sync recordings to fail. This will works for both, proxy and node recordings.
+func testSyncRecordingFailures(t *testing.T, suite *integrationTestSuite) {
```

Comment:
Is this test meaningfully different from the two below like `"proxy/node-sync recording requires auth connectivity"`?

Reply:
The other test validates the connect flow, ensuring the session doesn't start in case of auth failure.

Initially, I tried to do this test there, but eventually, there were too many options to deal with it, making the test somewhat complicated to follow.